pkg/apis/application/v1alpha1/types.go

```diff
@@ -702,6 +702,8 @@ type ApplicationCondition struct {
 	Type ApplicationConditionType `json:"type" protobuf:"bytes,1,opt,name=type"`
 	// Message contains human-readable message indicating details about condition
 	Message string `json:"message" protobuf:"bytes,2,opt,name=message"`
+	// Time is the time the condition was last set.
+	Time metav1.Time `json:"time"`
```

Comment:
big question, do we want to call it `lastTransitionTime` or summit to be similar to other resources?

Reply:
Yeah we typically use lastTransitionTime and lastProbeTime in our APIs. Am I correct that Argo clears conditions once not in use? If and without a status field, maybe lastProbeTime makes most sense here? Or perhaps both and set to same value for now?

Reply:
`lastProbeTime` - @jessesuen what do you think?

Reply:
Thinking about it some more I ended up on lastTransitionTime, based on @alexmt comments below about *not* updating the timestamp every time we check, it makes sense that this is the last time we transitioned to this state. (not the last time we checked, i.e. lastProbeTime) I think we if you were to start updating the timestamp every time we checked, you might want both variables at some point, but with the current goal below transition time seemed to make sense. Let me know if any concerns though and I can change it.